compiler/rustc_parse/src/parser/path.rs

```diff
@@ -880,6 +880,13 @@ impl<'a> Parser<'a> {
         &mut self,
         ty_generics: Option<&Generics>,
     ) -> PResult<'a, Option<GenericArg>> {
+        let mut attr_span: Option<Span> = None;
+        if self.token == token::Pound && self.look_ahead(1, |t| *t == token::OpenBracket) {
```

Comment:
Thinking about it, there's no reason not to include doc comments `///…␊` and `/**…*/` here and in parse_ty (i.e., getting rid of this `if` check and relying on `!is_empty` again) but the diagnostic on master isn't that bad, so not a blocking concern and can be done in a follow-up.